x-pack/legacy/plugins/code/public/reducers/repository_management.ts

```diff
@@ -93,18 +94,30 @@ export const repositoryManagement = handleActions<
         draft.importLoading = false;
         draft.showToast = true;
         draft.toastType = ToastType.success;
-        draft.toastMessage = `${action.payload!.name} has been successfully submitted!`;
+        draft.toastMessage = i18n.translate(
+          'xpack.code.repositoryManagement.repoSubmittedMessage',
+          {
+            defaultMessage: '{name} has been successfully submitted!',
+            values: { name: action.payload!.name },
```

Comment:
since name can be `undefined` here, i think it's a good idea to handle that gracefully in the translation. The message will be: `undefined has been successfully submitted!` here in some cases

Reply:
I believe why we do `action.payload!` previously is that `payload` can't be `undefined`, if it is undefined maybe we should bail out early and give same error logging, I'll talk to qianliang who original write this code on how we should handle it.